csrc/scheduler/transpose.cpp

```diff
@@ -127,6 +127,26 @@ void TransposeScheduler::computeHeuristics(
 
 namespace {
 
+// propagation could miss trivial reduction iterdomain on input tensors, since
+// that doesn't have any dependency and doesn't map to anything, we can naively
+// just reorder them so they won't interfere with tiling.
+// See https://github.com/NVIDIA/Fuser/issues/1659#issuecomment-1907053830
+void cleanInnerNDLeafDomain(TensorView* tv, int n) {
+  if (!tv->isFusionInput()) {
+    return;
+  }
+
+  std::unordered_map<int, int> old2new;
+
+  for (int i = 0; i < n; i++) {
+    if (tv->axis(-1 - i)->isReduction()) {
+      old2new[-1 - i] = i;
```

Comment:
Instead of making it `i`, should we have a counter of the number of reductions we have seen, and use that counter as index?

Reply:
oh I think that's what I was trying to do, but then forgot that when I change the code to take `n` as an argument :face_with_head_bandage: